src/app/extensions/quickorder/shared/direct-order/direct-order.component.ts

```diff
@@ -0,0 +1,122 @@
+
+@Component({
+  selector: 'ish-direct-order',
+  templateUrl: './direct-order.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+@GenerateLazyComponent()
+export class DirectOrderComponent implements OnInit, OnDestroy {
+  model = { sku: '', quantity: 1 };
+  directOrderForm = new FormGroup({
+    sku: new FormControl(this.model.sku),
+    quantity: new FormControl(this.model.quantity),
+  });
+  disabledAddToCart = true;
+  fields: FormlyFieldConfig[] = [
+    {
+      key: 'sku',
+      type: 'ish-text-input-field',
+      className: 'col-12 list-item search-container mb-1',
+      templateOptions: {
+        placeholder: 'shopping_cart.direct_order.item_placeholder',
+        autocomplete: true,
+      },
+      expressionProperties: {
+        'templateOptions.required': () => (!this.directOrderForm.untouched ? !!this.model.quantity : false),
+      },
+      asyncValidators: {
+        validProduct: {
+          expression: (control: FormControl) =>
+            control.valueChanges.pipe(
+              tap(() => {
+                this.checkDisabledStates(true);
+              }),
+              debounceTime(500),
+              switchMap(() => this.shoppingFacade.product$(control.value, ProductCompletenessLevel.List)),
+              tap(product => {
+                const failed = ProductHelper.isFailedLoading(product);
+                control.setErrors(failed ? { validProduct: false } : undefined);
+                this.checkDisabledStates(false);
+              }),
+              mapTo(undefined)
+            ),
+          message: () => this.translate.get('quickorder.page.error.invalid.product', { 0: this.model.sku }),
+        },
+      },
+      validation: {
+        messages: {
+          required: 'shopping_cart.direct_order.error.quantitywithoutsku',
+        },
+      },
+    },
+  ];
+
+  private destroy$ = new Subject();
+
+  disabled$: Observable<boolean>;
+
+  constructor(
+    private shoppingFacade: ShoppingFacade,
+    private translate: TranslateService,
+    private context: ProductContextFacade
+  ) {}
+
+  ngOnInit() {
+    this.setContext();
+    this.directOrderForm.valueChanges
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((control: { sku: string; quantity: number }) => {
+        this.context.set('sku', () => control.sku);
+        this.context.set('quantity', () => control.quantity);
+        this.model.quantity = control.quantity;
+        this.checkDisabledStates();
+      });
+    this.context
+      .select('quantity')
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(quantity => {
+        if (this.directOrderForm.get('quantity').value !== quantity) {
+          this.directOrderForm.get('quantity').setValue(quantity);
+        }
+      });
+  }
+
+  ngOnDestroy() {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  private setContext() {
+    this.context.set('sku', () => ' ');
```

Comment:
this is not working at all because empty sku s are currently filtered in the product context facade